src/lib/parallel_scan/dune

```diff
@@ -1,10 +1,10 @@
 (library
  (name parallel_scan)
  (public_name parallel_scan)
- (flags :standard -short-paths -warn-error -58)
+ (flags :standard -short-paths -warn-error -58-32)
```

Comment:
Would be really great to make both of these warnings errors again in this lib since you refactored it all.